app/models/project.server.ts

```diff
@@ -197,121 +210,143 @@ export async function getProjects(where: ProjectWhereInput) {
         skills: true,
         projectMembers: true,
       },
-    })
-    return projects
+    });
+    return projects;
   } catch (e) {
-    throw new Error(JSON.stringify(e))
+    throw new Error(JSON.stringify(e));
   }
 }
 
 export const validateIsTeamMember = (
   profileId: string,
-  data: { projectMembers: any[]; owner: { id: string } }
+  projectMembers: { profileId: string }[],
+  ownerId: string | null
 ) => {
   //validate if the user have permissions (team member or owner of the project)
-  const isProjectMember = data.projectMembers.some((member) => member.profileId === profileId)
-  const isProjectOwner = profileId === data.owner?.id
+  const isProjectMember = projectMembers.some(
+    (member) => member.profileId === profileId
+  );
+  const isProjectOwner = profileId === ownerId;
 
   if (!isProjectMember && !isProjectOwner) {
-    throw new Error("You don't have permission to perform this operation")
+    throw new Error("You don't have permission to perform this operation");
   }
-}
+};
 
-export async function updateProjects(
-  id: string,
+export async function updateMembers(
+  profileId: string,
   isAdmin: boolean,
-  data: {
-    projectMembers: any
-    existedMembers?: any
-    repoUrls?: any
-    relatedProjects?: any
-    projectStatus?: any
-    owner: any
-    skills?: any
-    disciplines?: any
-    labels?: any
-    innovationTiers?: any
-  }
+  projectId: string,
+  projectMembers: {
+    id?: string;
+    profileId: string;
+    hoursPerWeek?: number;
+    role?: { id: string }[];
+    practicedSkills?: { id: string }[];
+    active: boolean;
+  }[]
 ) {
   //validate if the user have permissions (team member or owner of the project)
-  if (!isAdmin) validateIsTeamMember(id, data)
+  const project = await db.projects.findUniqueOrThrow({
+    where: { id: projectId },
+    select: {
+      ownerId: true,
+    },
+  });
+  if (!isAdmin)
+    validateIsTeamMember(profileId, projectMembers, project.ownerId);
+
+  const previousMembers = await db.projectMembers.findMany({
+    where: { projectId },
+    select: { id: true, profileId: true },
+  });
 
-  let activeMembers: any = []
+  let activeMembers: any = [];
 
   // Loop Project Members
-  for (let index = 0; index < data.projectMembers.length; index++) {
-    // Creates the array for the practicedSkills connect
-    const practicedSkillsArrayConnect = data.projectMembers[index].practicedSkills?.map(
-      (skill: { id: any }) => {
-        return { id: skill.id }
-      }
-    )
-    // Creates the array for the roles connect
-    const rolesArrayConnect = data.projectMembers[index].role?.map((r: { id: any }) => {
-      return { id: r.id }
-    })
+  for (let i = 0; i < projectMembers.length; i++) {
     // Create only the members that don't exist in this project
-    if (data.projectMembers[index].profile) {
-      activeMembers.push(data.projectMembers[index]?.id)
+    const previousMember = previousMembers.find(
```

Comment:
This could be extracted to a function that has a name that reflects the intention like: createMembersThatDontExist
What do you think?